HLTrigger/Muon/src/HLTMuonL3PreFilter.cc

```diff
@@ -108,131 +110,244 @@ HLTMuonL3PreFilter::fillDescriptions(edm::ConfigurationDescriptions& description
+         if (absDxy > max_DXYBeamSpot_ || absDxy < min_DXYBeamSpot_ ) continue;
+  
+         //min muon hits cut
+         const reco::HitPattern& trackHits = tk->hitPattern();
+         if (trackHits.numberOfValidMuonHits() < min_NmuonHits_ ) continue;
+  
+         //pt difference cut
+         double candPt = cand->pt();
+         double trackPt = tk->pt();
+  
+         if (std::abs(candPt - trackPt) > max_PtDifference_ ) continue;
+  
+         //track pt cut
+         if (trackPt < min_TrackPt_ ) continue;
+  
+         // Pt threshold cut
+         double pt = cand->pt();
+         double err0 = tk->error(0);
+         double abspar0 = std::abs(tk->parameter(0));
+         double ptLx = pt;
+         // convert 50% efficiency threshold to 90% efficiency threshold
+         if (abspar0>0) ptLx += nsigma_Pt_*err0/abspar0*pt;
+         LogTrace("HLTMuonL3PreFilter") << " ...Muon in loop, trackkRef pt= "
+  				      << tk->pt() << ", ptLx= " << ptLx
+  				      << " cand pT " << cand->pt();
+        if (ptLx<min_Pt_) continue;
+  
+        filterproduct.addObject(TriggerMuon,cand);
+        n++;
+        break; // and go on with the next L2 association
+       }
+     }////loop over L2s from L3 grouping
+  
+     vector<RecoChargedCandidateRef> vref;
+     filterproduct.getObjects(TriggerMuon,vref);
+     for (unsigned int i=0; i<vref.size(); i++ ) {
+       RecoChargedCandidateRef candref =  RecoChargedCandidateRef(vref[i]);
+       TrackRef tk = candref->get<TrackRef>();
+       LogDebug("HLTMuonL3PreFilter")
+         << " Track passing filter: trackRef pt= " << tk->pt() << " (" << candref->pt() << ") " << ", eta: " << tk->eta() << " (" << candref->eta() << ") ";
+     }
+  
+     // filter decision
+     const bool accept (n >= min_N_);
+  
+     LogDebug("HLTMuonL3PreFilter") << " >>>>> Result of HLTMuonL3PreFilter is " << accept << ", number of muons passing thresholds= " << n;
+  
+     return accept;
+  } //end of useL3MTS
+
+  // Using normal TrajectorySeeds:
+  else{
+    LogDebug("HLTMuonL3PreFilter") << "HLTMuonL3PreFilter::hltFilter is in mode: not useL3MTS";
+
+    // Read Links collection:
+    edm::Handle<reco::MuonTrackLinksCollection> links;
+    iEvent.getByToken(linkToken_, links);
+
+    // Loop over RecoChargedCandidates:
+    for(unsigned int i(0); i < mucands->size(); ++i){
+      RecoChargedCandidateRef cand(mucands,i);
+      for(unsigned int l(0); l <links->size(); ++l){
+        const reco::MuonTrackLinks* link = &links->at(l);
+	bool useThisLink=false;
+	TrackRef tk = cand->track();
+	reco::TrackRef trkTrack = link->trackerTrack();
+
+	// Using the same method that was used to create the links
+	// ToDo: there should be a better way than dR,dPt matching
+	const reco::Track& globalTrack = *link->globalTrack();
+	float dR  = deltaR2(tk->eta(),tk->phi(),globalTrack.eta(),globalTrack.phi());
+	float dPt = std::abs(tk->pt() - globalTrack.pt())/tk->pt();
+	if (dR < 0.02 and dPt < 0.001) {
```

Comment:
Ah yes, I had changed it to deltaR2 after your previous comment:
https://github.com/cms-sw/cmssw/pull/7003#discussion-diff-22274958R282
Is it ok to go back to deltaR to keep it in line with the L3MuonCandidateProducer?